ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -353,21 +397,35 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    return Utils.hashcode(new Object[]{version, partitionId, uuid});
+    return uuid.hashCode();
   }
 
-  /**<p>
-   *   Gets the value of {@link #CURRENT_VERSION}.
-   * </p>
-   * <p>
-   *   It is typically not a good practice to call overridable methods from constructor. However, there could not be
-   *   cleaner way to set the {@link #version} field by either calling this method, or by reading from an input stream
-   *   from {@link BlobId#BlobId(String, ClusterMap)}, and also because this method simply returns a static final value
-   *   that does not depend on the state of an instance.
-   * </p>
-   * @return The value of {@link #CURRENT_VERSION}.
+  /**
+   * @return The version of this BlobId
    */
-  protected short getCurrentVersion() {
-    return CURRENT_VERSION;
+  protected short getVersion() {
+    return version;
+  }
+
+  /**
+   * @return all valid versions of BlobId.
+   */
+  public static Short[] getAllValidVersions() {
+    return new Short[]{BLOB_ID_V1, BLOB_ID_V2, BLOB_ID_V3};
+  }
+
+  /**
+   * Indicates the context in which a {@link BlobId} gets created.
+   */
+  public enum BlobIdType {
+    /**
+     * Indicates natively created BlobId (in the context of a PUT operation)
+     */
+    NATIVE, /**
```

Comment:
The autoformatter seems to have joined these lines. I usually just leave a blank line in between to satisfy it.

Reply:
Ah, didn't know adding a new line fixes this. Done.